OutOfSchool/OutOfSchool.AuthCommon/Services/IdentityRolesInitializerHostedService.cs

```diff
@@ -0,0 +1,102 @@
+﻿using Microsoft.Extensions.Hosting;
+
+namespace OutOfSchool.AuthCommon.Services;
+
+/// <summary>
+/// Hosted service that initializes default roles for correct work of authorization server.
+/// </summary>
+public class IdentityRolesInitializerHostedService : IHostedService
+{
+    private readonly IServiceProvider serviceProvider;
+    private readonly ILogger<IdentityRolesInitializerHostedService> logger;
+
+    public IdentityRolesInitializerHostedService(
+        IServiceProvider serviceProvider,
+        ILogger<IdentityRolesInitializerHostedService> logger)
+    {
+        this.serviceProvider = serviceProvider;
+        this.logger = logger;
+    }
+
+    /// <summary>
+    /// Get default role names that should be initialized for correct work of authorization server.
+    /// </summary>
+    /// <returns>New array that contains default role names.</returns>
+    public static string[] GetDefaultRoleNames() =>
+    [
+        "parent",
+        "provider",
+        "techadmin",
+        "ministryadmin",
+        "regionadmin",
+        "areaadmin",
+    ];
+
+    public async Task StartAsync(CancellationToken cancellationToken)
+    {
+        using var scope = serviceProvider.CreateScope();
+
+        string[] defaultRoleNames = GetDefaultRoleNames();
+
+        var roleManager = scope.ServiceProvider.GetRequiredService<RoleManager<IdentityRole>>();
+
+        var existingRoleNames = await roleManager.Roles
+            .Where(role => defaultRoleNames.Contains(role.Name))
+            .Select(role => role.Name)
+            .ToArrayAsync(cancellationToken)
+            .ConfigureAwait(false);
+
+        var newRoles = defaultRoleNames
+            .Except(existingRoleNames)
+            .Select(roleName => new IdentityRole { Name = roleName });
```

Comment:
(Logic, performance) In selected code you are filtering roles twice: remove from existing all that are not default, and then remove from default all that are not existing. It takes excessive code and CPU. 
Instead you could make filtering one time - take all existing role names from roleManager (not filtered) and exclude them from default.